src/main/java/frc/robot/subsystems/SwerveSubsystem.java

```diff
@@ -224,6 +233,8 @@ public Command sysIdDynamic(SysIdRoutine.Direction direction) {
 
   @Override
   public void periodic() {
+    robotChassisSpeeds = calculateCurrentRobotChassisSpeeds();
```

Comment:
Same pattern of "calculate once, then reuse". This minimizes GC pressure caused from getting robot chassis speeds, especially here for logging.